File: src/main/java/org/tobi29/scapes/engine/gui/GuiRenderer.java
package org.tobi29.scapes.engine.gui;

import org.tobi29.scapes.engine.graphics.Matrix;
import org.tobi29.scapes.engine.graphics.MatrixStack;
import org.tobi29.scapes.engine.utils.math.vector.Vector3;
import org.tobi29.scapes.engine.utils.math.vector.Vector3f;

public class GuiRenderer extends GuiRenderBatch {
    private final MatrixStack matrixStack = new MatrixStack(64);

    public MatrixStack matrixStack() {
        return matrixStack;
    }

    @Override
    public Vector3 vector(float x, float y) {
        Matrix matrix = matrixStack.current();
        return matrix.modelView().multiply(new Vector3f(x, y, 0.0f));
    }
}

File: Engine/src/main/kotlin/org/tobi29/scapes/engine/gui/GuiRenderer.kt
/*
 * Copyright 2012-2017 Tobi29
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.tobi29.scapes.engine.gui

import org.tobi29.scapes.engine.graphics.MatrixStack
import org.tobi29.math.vector.Vector2d
import org.tobi29.math.vector.Vector3d

class GuiRenderer : GuiRenderBatch(Vector2d(1.0, 1.0)) {
    private val matrixStack = MatrixStack(64)

    fun matrixStack(): MatrixStack {
        return matrixStack
    }

    override fun vector(x: Double,
                        y: Double): Vector3d {
        val matrix = matrixStack.current()
        return matrix.modelView().multiply(Vector3d(x, y, 0.0))
    }
}
